Purchases/Networking/Operations/PostOfferForSigningOperation.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PostOfferForSigningOperation.swift
+//
+//  Created by Joshua Liebowitz on 11/19/21.
+
+import Foundation
+
+class PostOfferForSigningOperation: NetworkOperation {
+
+    private let configuration: UserSpecificConfiguration
+    private let offerIdentifier: String
+    private let productIdentifier: String
+    private let subscriptionGroup: String
+    private let receiptData: Data
+    private let completion: OfferSigningResponseHandler
+
+    init(configuration: UserSpecificConfiguration,
+         offerIdForSigning offerIdentifier: String,
+         productIdentifier: String,
+         subscriptionGroup: String,
+         receiptData: Data,
+         completion: @escaping OfferSigningResponseHandler) {
+        self.configuration = configuration
+        self.offerIdentifier = offerIdentifier
+        self.productIdentifier = productIdentifier
+        self.subscriptionGroup = subscriptionGroup
+        self.receiptData = receiptData
+        self.completion = completion
+
+        super.init(configuration: configuration)
+    }
+
+    override func main() {
+        if self.isCancelled {
+            return
+        }
+
+        self.post(offerIdForSigning: self.offerIdentifier,
+                  productIdentifier: self.productIdentifier,
+                  subscriptionGroup: self.subscriptionGroup,
+                  receiptData: self.receiptData,
+                  appUserID: self.configuration.appUserID,
+                  completion: self.completion)
+    }
+
+    // swiftlint:disable:next function_parameter_count function_body_length
+    func post(offerIdForSigning offerIdentifier: String,
+              productIdentifier: String,
+              subscriptionGroup: String,
+              receiptData: Data,
+              appUserID: String,
+              completion: @escaping OfferSigningResponseHandler) {
+        let fetchToken = receiptData.asFetchToken
+
+        let requestBody: [String: Any] = ["app_user_id": appUserID,
```

Comment:
maybe this one could all be a struct similar to `PostData` in `PostReceiptDataOperation`?

Reply:
Yes! Definitely!

Reply:
Updated! I encapsulated it 😄